app/frontend/src/components/TabBar/TabBar.tsx

```diff
@@ -29,20 +31,23 @@ export default function TabBar<T extends Record<string, React.ReactNode>>({
   return (
     <TabList
       aria-label={ariaLabel}
+      value={value}
       onChange={handleChange}
       indicatorColor="primary"
       textColor="primary"
       scrollButtons="auto"
       variant="scrollable"
     >
-      {Object.entries(labels).map(([value, label]) => (
-        <Tab
-          key={value}
-          label={label}
-          value={value}
-          className={classes.messagesTab}
-        />
-      ))}
+      {Object.entries(labels)
+        .filter(([isDefined]) => !!isDefined)
```

Comment:
```suggestion
```
I'm not sure there's a good case for supporting undefined labels, since we can always add to the tab label later when we actually have content for it.